hazelcast/src/main/java/com/hazelcast/json/internal/JsonSchemaStructNode.java

```diff
@@ -89,7 +98,38 @@ public int hashCode() {
     @Override
     public String toString() {
         return "JsonSchemaStructNode{"
-                + "inners=" + inners
-                + '}';
+            + "inners=" + inners
+            + '}';
+    }
+
+    @Override
+    public void writeData(ObjectDataOutput out) throws IOException {
+        out.writeInt(inners.size());
+        for (JsonSchemaNameValue nameValue : inners) {
+            out.writeObject(nameValue);
+        }
+        // Don't serialize parent node from superclass to avoid cyclic dependency
+    }
+
+    @Override
+    public void readData(ObjectDataInput in) throws IOException {
+        int innersSize = in.readInt();
+        inners = new ArrayList<>(innersSize);
+        for (int i = 0; i < innersSize; ++i) {
+            JsonSchemaNameValue nameValue = in.readObject();
+            nameValue.getValue().setParent(this);
+            inners.add(nameValue);
+        }
+    }
+
+    @Override
+    public int getFactoryId() {
+        return MapDataSerializerHook.F_ID;
```

Comment:
maybe we can introduce `JsonDataSerializerHook`, seems this json package was thought to be kept in isolation with other packages for general purpose usages.

Reply:
Addressed in https://github.com/hazelcast/hazelcast/pull/18186/commits/5ddd8c32715b5b231dea0c4f64108b2b75e213d2